ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -428,6 +439,18 @@ public void setFontStyle(@Nullable String fontStyleString) {
     }
   }
 
+  @ReactProp(name = ViewProps.TEXT_DECORATION_LINE)
+  public void setTextDecorationLine(@Nullable String textDecorationLineString) {
+    for(String textDecorationLineSubString:textDecorationLineString.split(" ")) {
+      if ("underline".equals(textDecorationLineSubString)) {
```

Comment:
If we have new data for text decoration, we should reset the two flags. Otherwise, setting 'line-through' after 'underline line-through' will not clear the underline flag.